components/net_traits/response.rs

```diff
@@ -75,6 +75,19 @@ pub enum ResponseMsg {
     Errored,
 }
 
+#[derive(Serialize, Deserialize, Clone)]
+pub struct ResponseInit {
+    pub url: ServoUrl,
```

Comment:
We will need more data here; for example, http_redirect_fetch looks at the headers of the response, and those will be initialized as empty in this case. The FetchMetadata type is serializable and contains everything we should need, I believe.